src/pudl/transform/ferc714.py

```diff
@@ -319,6 +340,270 @@ def _pre_process(df: pd.DataFrame, table_name: str) -> pd.DataFrame:
     return out_df
 
 
+def melt_hourx_columns_csv(df):
+    """Melt hourX columns into hours."""
+    # Almost all 25th hours are unusable (0.0 or daily totals),
+    # and they shouldn't really exist at all based on FERC instructions.
+    df = df.drop(columns="hour25")
+
+    # Melt daily rows with 24 demands to hourly rows with single demand
+    logger.info("Melting daily FERC 714 records into hourly records.")
+    df = df.rename(
+        columns=lambda x: int(re.sub(r"^hour", "", x)) - 1 if "hour" in x else x,
+    )
+    df = df.melt(
+        id_vars=[
+            "respondent_id_ferc714",
+            "report_year",
+            "report_date",
+            "utc_offset_code",
+        ],
+        value_vars=range(24),
+        var_name="hour",
+        value_name="demand_mwh",
+    )
+    return df
+
+
+def map_respondent_id_ferc714(df, source: Literal["csv", "xbrl"]):
+    """TODO: Make this map!"""
+    return df
```

Comment:
right now the table gets fully through the transforms except for the enforce_schema step because the respondent_id_ferc714 now contains both the old int id and the new string id. we need to build the map and replace them here.